controlers/auth.js

```diff
@@ -0,0 +1,124 @@
+const User = require('../models/User');
+const Joi = require('joi');
+const jwt = require('jsonwebtoken');
+
+const userJoi = Joi.object({
+    password: Joi.string().min(3).max(30).required(),
+    email: Joi.string().email().required()
+});
+
+const register = async (req, res, next) => {
+    const { password, email } = req.body
+    const user = await User.findOne({ email })
+    if (user) {
+        return res.status(409).json({
+            status: 409,
+            message: 'Email in use',
+        })
+    };
+    const { error } = userJoi.validate(req.body);
```

Comment:
Zrób to na samym początku kodu, czyli kiedy robisz 
```
const {password, email} = req.body;
```
a dopiero potem sprawdzaj czy taka osoba istnieje. Bo nie wiesz czy email który ktoś wysłał jest poprawny. To jest zasada do wszystkich endpointów gdzie robisz walidację req.body.

Reply:
poprawione